src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -1526,7 +1526,12 @@ private bool CompleteScriptBlockBody(Token lCurly, ref IScriptExtent bodyExtent,
 
                     UngetToken(rCurly);
                     endScriptBlock = bodyExtent ?? lCurly.Extent;
-                    ReportIncompleteInput(lCurly.Extent, rCurly.Extent, () => ParserStrings.MissingEndCurlyBrace);
+                    Expression<Func<string>> errorExpr = () => ParserStrings.MissingEndCurlyBrace;
+                    if (calledFromNamedBlockRule)
```

Comment:
Maybe use `calledFromNamedBlockRule ? ParserStrings.MissingEndCurlyBraceOrNamedBlock: ParserStrings.MissingEndCurlyBrace`

Reply:
The `Expression<Func<string>>` passed in needs to have a body of `MemberExpression` for the parser to get the correct error id. See [`SaveError`](https://github.com/daxian-dbw/PowerShell/blob/24599fa413485ad65738f606bd06c97b727bc3f3/src/System.Management.Automation/engine/parser/Parser.cs#L6837).